apisix/discovery/nacos.lua

```diff
@@ -0,0 +1,237 @@
+    return httpc:request_uri(url, {
+        version = 1.1,
+        method = method,
+        headers = headers,
+        query = query,
+        body = body,
+        ssl_verify = false,
+    })
+end
+
+
+local function get_url(request_url,basic_auth,path)
+    local res, err = request(request_url, basic_auth, "GET", path)
+    if not res then
+        log.error("failed to fetch registry", err)
+        return
+    end
+
+    if not res.body or res.status ~= 200 then
+        log.error("failed to fetch registry, status = ", res.status)
+        return
+    end
+
+    local json_str = res.body
+    local data, err = core.json.decode(json_str)
+    if not data then
+        log.error("invalid response body: ", json_str, " err: ", err)
+        return
+    end
+    return data
+end
+
+
+local function fetch_full_registry(premature)
+    if premature then
+        return
+    end
+
+    local base_uri, basic_auth = service_info()
+    if not base_uri then
+        return
+    end
+
+    local up_apps = core.table.new(0, 0)
+    local data = get_url(base_uri,basic_auth,service_list_path)
+    if tostring(data.count) == "0" then
+        applications = up_apps
+        return
+    end
+
+    for _, service_name in ipairs(data.doms) do
+        data = get_url(base_uri,basic_auth,instance_list_path .. service_name)
+        for _, host in ipairs(data.hosts) do
+            if tostring(host.valid) == 'true' and
+                    tostring(host.healthy) == 'true' and
+                    tostring(host.enabled) == 'true' then
+                local nodes = up_apps[service_name]
+                if not nodes then
+                    nodes = core.table.new(0, 0)
+                    up_apps[service_name] = nodes
+                end
+                core.table.insert(nodes, {
+                    host = host.ip,
+                    port = host.port,
+                    weight = host.weight or default_weight,
+                })
+            end
+        end
+    end
+    applications = up_apps
+end
+
+
+function _M.nodes(service_name)
+    if not applications then
+        log.error("failed to fetch nodes for : ", service_name)
+        return
+    end
+
+    return applications[service_name]
+end
+
+
+function _M.init_worker()
+    if not local_conf.discovery.nacos or
+            not local_conf.discovery.nacos.host or #local_conf.discovery.nacos.host == 0 then
+        error("do not set nacos.host")
+        return
+    end
+
+    local ok, err = core.schema.check(schema, local_conf.discovery.nacos)
+    if not ok then
+        error("invalid nacos configuration: " .. err)
+        return
+    end
+    default_weight = local_conf.discovery.nacos.weight or 100
+    log.info("default_weight:", default_weight, ".")
+    local fetch_interval = local_conf.discovery.nacos.fetch_interval or 30
+    log.info("fetch_interval:", fetch_interval, ".")
+    service_list_path = local_conf.discovery.nacos.service_list_path or
```

Comment:
Where should I configure `service_list_path`? There is neither schema nor doc for it.

Reply:
I will fix it.